ghconfig.py

```diff
@@ -69,20 +67,43 @@ def remove_branch_protection(organisation, repository, branch):
     return response.status_code
 
 
+def load_configuration(configuration_file):
+    with open(configuration_file) as config_file:
+        configuration = yaml.load_safe(config_file)
+        return configuration
+
+
 @click.option(
     '--organisation', help='Name of the Organisation or GitHub user.',
     default='alphagov')
 @click.option(
-    '--filter', help='Specify a filter for repository names (example: paas-'
+    '--filter', help='Specify a filter for repository names (example: paas-',
     default='')
+@click.option(
+    '--configuration', help='Configuration file', default='ghconfig.yml')
 @click.command()
-def check_config(organisation, filter):
+def check_config(organisation, filter, configuration):
     print('Checking GitHub configuration...')
 
     # Filter repositories to work on using the specified organisation and
     # repository names filter
-    repos = search_repositories(organisation, filter)
-
+    repositories = search_repositories(organisation, filter)
+
+    # Load configuration from file
+    config = load_configuration(configuration)
+
+    # Apply the configuration to found braches
+    for repository in repositories:
+        if config.get('protected_branches'):
+            for branch in config['protected_branches']:
+                print(
+                    'Protecting branch {0} on {1}/{2}'.format(
+                        branch, organisation, repository['name']))
+                set_branch_protection(
```

Comment:
I've tried on my own, it behaves like if it wasn't protected and set the value again. Basically like setting a variable to "True" when it's already True.